### ISSUE TITLE
As a editor I want to be able to crop the images inside the text editor of news articles to avoid image manipulation outside the studio

### ISSUE THREAD

**Issue by wenche** [equinor/energyvision#256]
## Acceptance criteria
1. For inline images the image should be cropped, but it shouldn't use hotspot features

## In order to ensure good quality we should (Keep the ones we need for this US)
- [ ] Perform integration testing
- [ ] Perform functional testing
- [ ] Extra focus on usability
- [ ] Test for portability
- [ ] Test for accessibility
- [ ] Create automated acceptance test/smoke test
- [ ] Perform performance and load test
- [ ] Perform user testing
- [ ] Perform security testing

**Comment by wenche** [comment]
TL;DR: Images in a responsive world are hard.

For the rich text editor we don't know the aspect ratio upfront as the images can be things like figures, maps or portraits of people. That means we have to display the image as it is with unknown height, unlike the hero image and cards images with predefined aspect ratio. This will cause issues with CLS (Cumulative layout shift) since the browser can't set aside space in advance. The `next-sanity-image` package and Next.js image pipeline work by mandatory predefined height and width, where height with aspect ratio is calculated. 

With the images inside the rich text editor this is not possible, and the way we initially created these images were based on the original dimensions on `width` and `height` without the crop and highlight information from Sanity.

The way the image information is saved in Sanity is by an object containing the asset itself with original dimension, the hotspot info and crop info.
```
image {
  asset: {...},
  crop: {....},
  hotspot: {...},
}
```
We just passed the information in image.asset to the image pipeline, and this works well for images without aspect ratio by using the original dimensions. To support the `crop` feature however, we needed to pass the `crop` information as well, and that is far more complicated.  Out of the box this will result in an image that is stretched to fill the dimension box based on original width and height (since the crop information is applied after the size calculations):

![Skjermbilde 2021-04-26 kl. 09.40.48.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/b37dc407-84db-45a5-a5ab-ad417fc7e547)

We could use `object-fit: contain` for the `<img>` tag and then the result will be an image that is actually cropped that means it's smaller than the available width/height. The pros is that the image will look good, just smaller.

![Skjermbilde 2021-04-26 kl. 09.42.50.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/b1097b85-e48c-4025-b87b-dcbb338d3f53)

While this might be acceptable, it's not a 100% perfect solution. The conclusion for this issue is that we need to investigate this a lot more to find a better solution.
 **Some ideas and things to remember:**
- Since the `next-sanity-image` library only supports the `responsive` layout mode could we fork this library and do something more clever for other types of images?
- We want the lazy loading features of the Next.js image pipeline as well as using the Sanity image API
- We could generate image url with the API ourselves but then we'll need the lazy loading as a separate library
- If we are able to find the correct image dimensions for the cropped version and not the original version this will be a whole lot easier

**Plan:** Create a separate issue and look more into this at a later stage. Watch Next.js image pipeline for new features. In the meantime, roll back the cropping functionality as we don't have a final solution with regards on both performance , UX and publisher's experience yet.
